### ISSUE TITLE
K3s won't start with Docker runtime

### ISSUE THREAD

**Issue by chriswgerber** [abiosoft/colima#759]
### Description

Starting up colima with Kubernetes using the default runtime fails to load metrics server, and in turn, all pods fail to start up. Switching to containerd enables metrics server to start up and pods to become available.

### Version

Colima Version:
```
colima version 0.5.5
git commit: 6251dc2c2c5d8197c356f0e402ad028945f0e830

runtime: docker
arch: aarch64
client: v24.0.4
server: v23.0.6

kubernetes
Client Version: v1.27.3
Kustomize Version: v5.0.1
Server Version: v1.24.3+k3s1
```

Lima Version:
```
limactl version 0.16.0
```

Qemu Version:
```
qemu-img version 8.0.3
Copyright (c) 2003-2022 Fabrice Bellard and the QEMU Project developers
```

### Operating System

- [ ] macOS Intel <= 12 (Monterrey)
- [ ] macOS Intel >= 13 (Ventura)
- [ ] macOS M1 <= 12 (Monterrey)
- [X] macOS M1 >= 13 (Ventura)
- [ ] Linux

### Output of `colima status`

```
INFO[0000] colima is running using QEMU
INFO[0000] arch: aarch64
INFO[0000] runtime: docker
INFO[0000] mountType: sshfs
INFO[0000] socket: unix:///Users/-/.colima/default/docker.sock
INFO[0000] kubernetes: enabled
```

### Reproduction Steps

1. `colima start --kubernetes --cpu 4 --memory 8 --disk 50`
2. `kubectl get pods -A`
3. `grep "\-\-docker" /var/log/k3s.log`

### Additional context

`colima ssh grep "\-\-docker" /var/log/k3s.log`

Returns 

```
time="" level=info msg="Waiting to retrieve agent configuration; server is not ready: --docker is no longer supported; to continue using docker, install cri-dockerd and set --container-runtime-endpoint"

**Comment by abiosoft** [comment]
I have reproduced this and working on a fix

**Comment by chriswgerber** [comment]
Seems to be just an issue with K3s versions 1.24.1, 1.24.2, and 1.24.3: https://github.com/rancher-sandbox/rancher-desktop/issues/3286

**Comment by abiosoft** [comment]
Yeah, it is fixed as part of #757 and would be part of the next release.

**Comment by chriswgerber** [comment]
Added `--kubernetes-version "v1.27.1+k3s1"` to the start command and pods came up successfully.
